sdk/daml-lf/transaction/src/main/scala/com/digitalasset/daml/lf/value/CidContainer.scala

```diff
@@ -42,13 +42,15 @@ trait CidContainer[+A] {
     }
   }
 
-  // Sets the suffix of any the V1 ContractId `coid` of the container that are not already suffixed.
-  // Uses `f(coid.discriminator)` as suffix.
-  final def suffixCid(f: crypto.Hash => Bytes): Either[String, A] =
+  // Sets the suffix of any V1 or V2 ContractId `coid` of the container that are not already suffixed.
+  // Uses `f1(coid.discriminator.bytes)` or `f2(coid.local)` as suffix.
+  final def suffixCid(f1: crypto.Hash => Bytes, f2: Bytes => Bytes): Either[String, A] =
```

Comment:
as a side note, I would I prefer to keep the function with only one argument, but this will require more changes so it is probably fine like that.

Reply:
I had also thought about this, but it just meant to either declare a union type for the two formats or resort to `Either`. I'll leave this as is for now; we can refactor this when the two arguments become too annoying.